src/helpers/parseVAST.js

```diff
@@ -13,24 +13,29 @@ export default async function parseVAST(src, options = DEFAULT_VAST_OPTIONS) {
     const parsedVAST = await vastClient.get(srcWithSupportedMacros, options);
     this.VASTData = parsedVAST;
     const ad = parsedVAST?.ads[0];
-    const linearCreative = ad?.creatives?.find(({ type }) => type === 'linear');
-    if (!linearCreative) return;
-
-    const { mediaFiles = [] } = linearCreative;
-    // Separate any JavaScript files from MediaFiles
-    const sources = mediaFilesReducer(mediaFiles);
-    this.VASTSources = sources;
-    const VPAIDSource = sources.javascript.find(({ apiFramework }) => apiFramework === 'VPAID');
+    const creative = ad?.creatives[0];
+    if (!creative) return;
 
+    let VPAIDSource;
+    if (creative.type === 'linear') {
+        const { mediaFiles = [] } = creative;
+        // Separate any JavaScript files from MediaFiles
+        const sources = mediaFilesReducer(mediaFiles);
+        this.VASTSources = sources;
+        VPAIDSource = sources.javascript.find(({ apiFramework }) => apiFramework === 'VPAID');
+    } else if (creative.type === 'nonlinear') {
+        VPAIDSource = {
+            fileURL: creative?.variations[0].staticResource
```

Comment:
Is this always going to be a VPAID resource? With linear we usually check the apiFramework attribute to see if it's VPAID or something else.

Reply:
nvm, I just realized this is not applicable for non-linear